src/Symfony/Component/HttpClient/ThrottlingHttpClient.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpClient;
+
+use Symfony\Component\RateLimiter\LimiterInterface;
+use Symfony\Component\RateLimiter\RateLimiterFactory;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+use Symfony\Contracts\Service\ResetInterface;
+
+class ThrottlingHttpClient implements HttpClientInterface, ResetInterface
+{
+    use DecoratorTrait {
+        reset as private traitReset;
+    }
+
+    private LimiterInterface $rateLimiter;
+
+    public function __construct(
+        HttpClientInterface $client,
+        private readonly RateLimiterFactory $rateLimiterFactory,
+    ) {
+        $this->client = $client;
+    }
+
+    public function request(string $method, string $url, array $options = []): ResponseInterface
+    {
+        $response = $this->client->request($method, $url, $options);
+
+        if (0 < $waitDuration = $this->rateLimiter()->reserve()->getWaitDuration()) {
+            $response->getInfo('pause_handler')($waitDuration);
+        }
+
+        return $response;
+    }
+
+    private function rateLimiter(): LimiterInterface
+    {
+        return $this->rateLimiter ??= $this->rateLimiterFactory->create();
+    }
+
+    public function reset(): void
```

Comment:
Where is this method registered? If it's not, what about removing it? I'm not sure we need it: the factory already has a reset method, that's the one that should be used when needing a reset to me.

Reply:
Did you mean the `LimiterInterface`? It has a `reset()` method, but as far as I can tell, it needs to be called manually.

Reply:
I mean AbstractRequestRateLimiter::reset, which implements what's needed.

Reply:
I also mean this might be considered as an external concern from the pov of ThrottlingHttpClient - it doesn't need to care of the rate limiter is resettable or not.

Reply:
Would it make sense to pass a LimiterInterface directly then instead of a factory?

Reply:
Not sure it would make a difference, the limiter itself is not registered as a service, only the factory is, so we would still need to call `reset()` manually.

Reply:
bu the implementation would be less hacky: $this->limiter->reset() would do it

Reply:
Done https://github.com/symfony/symfony/compare/84c892749c3555a2c5f9fc0d18cae19e8d2064c9..1a8b068d6149cc40631dd9136f2a3279bbd127d1